Update devpi

nbro10 opened the pull request:
Hopefully, solves https://github.com/wbrp/deploy-devpi/issues/4

Do we need [this file](https://github.com/wbrp/docker-devpi/blob/update-devpi/devpi/upgrade.sh)? If not, I can delete it and update the `Dockerfile` not to copy it into the container.

Note: 

1. This PR removes an image (`data`), which I think was unnecessary because it just used to create a folder, but we can do it in the `devpi` image
2. This PR changes the base images from `alpine:3.4` to `python:3.8.14-slim` because I was getting an error installing `devpi-server` on that image. I think we could solve this problem by installing more dependencies, like [here](https://github.com/dmartin35/docker-devpi/blob/master/Dockerfile#L11), but, after having done that, it seemed like the image was larger than `python:3.8.14-slim`, but I could be wrong because I haven't investigated this much

Commit:
Update devpi versions and refactor docker code

Commit:
Update README.md

Commit:
Update devpi/Dockerfile

Commit:
Update requirements.txt with the latest versions

Commit:
Clarify info in README

ddoskind commented:
@nbro10 I think the `upgrade.sh` file is indeed no longer used as devpi now seems to have the ability to upgrade in-place. Feel free to add a commit removing it.

nbro10 commented:
@ddoskind So, probably, we also don't need the folder `/devpi/server-upgrade` inside the container, right? I am not sure. This folder is mapped to a volume and it may contain some files already.

Commit:
Remove upgrade.sh as no longer needed

ddoskind commented:
> @ddoskind So, probably, we also don't need the folder `/devpi/server-upgrade` inside the container, right? I am not sure. This folder is mapped to a volume and it may contain some files already.

I am almost certain this is correct, yes.

Commit:
Remove server-upgrade folder and volume

nbro10 commented:
@ddoskind I removed the `/devpi/server-upgrade` folder inside the container and the corresponding volume. Let me know if I should change something else. 

Btw, idk if you noticed, but I changed the `docker-compose.yml` to use [named volumes](https://github.com/wbrp/docker-devpi/blob/update-devpi/docker-compose.yml#L18), while, previously, we were using anonymous volumes. I wonder how the saved anonymous volume is used instead of the named one. `ls /var/lib/docker/volumes` inside `deploy-devpi` does not show the named volumes but `docker volume ls` does.

ddoskind commented:
ACK until next week 👍

ddoskind commented on .gitignore:
```diff
@@ -1,2 +1,191 @@
-*.swp
-dump-*
+# MacOS .gitignore
```
This is very comprehensive, that's why I'd recommend using it as a global .gitignore file and only add those entries in project repositories which you expect to be relevant (e.g. a `VENV` directory when there's makefiles that create one, etc.)

ddoskind commented on README.md:
```diff
@@ -1,88 +1,77 @@
 
-This Docker Compose based setup runs DevPi behind Nginx.
+This repository contains a [`docker-compose.yml`](./docker-compose.yml), which can be used to build
+2 docker images from 2 services, `devpi` and `nginx`, which share 2 named volumes `server` and 
+`server-upgrade`. 
```
This should be adjusted as `server-upgrade` is no longer required.

ddoskind reviewed (changes requested):
Perfectly fine, but please slim down the `.gitignore` file a bit and remove the reference to the `server-upgrade` volume from the docs 👍

Commit:
IFFCR

ddoskind reviewed (approved):
🌈  Merge away
